src/index.js

```diff
@@ -147,17 +167,46 @@ function toVdom(element, nodeName) {
 	}
 
 	for (i = cn.length; i--; ) {
-		const vnode = toVdom(cn[i], null);
+		const vnode = toVdom(cn[i], null, shadow);
 		// Move slots correctly
 		const name = cn[i].slot;
 		if (name) {
-			props[name] = h(Slot, { name }, vnode);
+			props[name] = h(
+				Slot,
+				{
+					name,
+					shadow,
+					addContextListener(listener, element = cn[i]) {
```

Comment:
This doesn't work as expected. When called via `addContextListener(this._listener);` (see Line 148) `element` is undefined.
I fixed this by defining a `const childElement = cn[i];` at the beginning of the `for` loop and using `childElement` instead of `cn[i]`.

Reply:
As far as I can see this seems to be a transpiler (in my case rollup) bug... 

Rollup transpiles from 

```
let i = 0;
for (i = 1; i--;) {
  do_something_with(cn[i])
}
``` 

to code like

```
var i = 0;
function _loop() {
  do_something_with(cn[i])
}
for (i = 1; i--;) {
  loop();
}
``` 

Which doesn't make any sense....